msteams-platform/messaging-extensions/how-to/search-commands/respond-to-search.md

```diff
@@ -91,9 +91,55 @@ Your service must respond with the results matching the user query. The response
 |`composeExtension.suggestedActions`|Suggested actions. Used for responses of type `auth` or `config`. |
 |`composeExtension.text`|Message to display. Used for responses of type `message`. |
 
-### Configuration response
+### `config` response type
 
-Configuration response is the data returned by the server or application to configure and enable the message extension within the messaging platform. The following code is an example for message extension configuration:
+The configuration response is the data returned by the server or application to configure and enable the message extension within the messaging platform. A `config` response is used when a user configures the message extension for the first time. 
+It triggers a prompt for the user to set up the message extension and provide any necessary configuration.
+
+If a message extension uses a configuration page, the [`onQuery`](/dotnet/api/microsoft.teams.ai.messageextensions-1.onquery) handler should first verify for any stored configuration data. If the message extension isn't configured, a `config` response must be returned, which includes a link to your configuration.
+
+The [`onQuery`](/dotnet/api/microsoft.teams.ai.messageextensions-1.onquery) handler also manages the response from the configuration page. The `onQuerySettingsUrl` handler provides the URL for the configuration page. After closing the configuration page, the `onSettingsUpdate` method is called to accept and save the returned state. The [`onQuery`](/dotnet/api/microsoft.teams.ai.messageextensions-1.onquery) handler then retrieves the updated settings and uses them to update the behavior of the message extension.
+
+The following code handles a user request for the configuration page of a message extension. It fetches the user's existing configuration settings, escapes them, and builds a response. This response includes the configuration page's URL with the escaped settings added as a query parameter.
```

Comment:
I think this section is conflating the config response type with the ME settings protocol. They're similar, but not the same.

Reply:
This should mention the canUpdateConfiguration property in the manifest and how that interacts with ME settings.

Reply:
<img width="557" alt="image" src="https://github.com/user-attachments/assets/34b647b9-e939-4203-a606-0644a577f752">

"Nihari, canUpdateConfiguration is used when user right clicks on an app in the ME flyout to check whether the app has a settings scenario. The query onQuerySettingsUrl is used to verify that. The 'config' response is send when responding to the ME query command. I think it is better to put the settings URL scenario in a different section."